### PR TITLE
HPCC-16038 Ensure headers and separators are utf8 for unicode csv

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

File: ecl/hqlcpp/hqlhtcpp.cpp [hunk 1]
```diff
@@ -9542,15 +9540,51 @@ void HqlCppTranslator::buildCsvListFunc(BuildCtx & classctx, const char * func,
...
+            ForEachChild(i, expr)
+            {
+                IHqlExpression * value = expr->queryChild(i);
+                args.append(*forceToCorrectEncoding(value, encoding));
+            }
+            return expr->clone(args);
+        }
+        else
+        {
+            OwnedHqlExpr cast = ensureExprType(expr, unknownUtf8Type);
```
Comment (shamser) [inline ecl/hqlcpp/hqlhtcpp.cpp]: What's unknownUtf8Type?  Not sure I understand how UTF-8 type can be an unknown type.  Should this be unknownUnicodeType?

**Review by shamser — COMMENTED** [review]
@ghalliday Have a question regarding UnknownUTf-8Type.. everything else looks good.

**Comment by ghalliday** [comment]
@shamser unknown refers to unknown length - so it creates a string based on the length of the expression.

**Pull request merged**